src/common/config.cc

```diff
@@ -600,14 +600,18 @@ Config::Config(std::string jsonfilename)
       "Demodulation block size must be a multiple of transpose block size");
   demul_events_per_symbol_ = 1 + (ofdm_data_num_ - 1) / demul_block_size_;
 
-  beam_batch_size_ = tdd_conf.value("beam_batch_size", 1);
-  beam_block_size_ = freq_orthogonal_pilot_
-                         ? pilot_sc_group_size_
-                         : tdd_conf.value("beam_block_size", 1);
-  beam_block_active_sc_ = freq_orthogonal_pilot_
-                              ? 1  // only compute one sc per call
-                              : beam_block_size_;
-  beam_calls_per_symbol_ = 1 + (ofdm_data_num_ - 1) / beam_block_size_;
+  beam_block_size_ = tdd_conf.value("beam_block_size", 1);
+  if (freq_orthogonal_pilot_) {
+    //Set beam block size to the pilot sc group size so events arn't generated for the redundant sc
+    if (beam_block_size_ != 1) {
+      AGORA_LOG_WARN(
+          "Setting beam block size to %zu because freq orthogonal pilot is "
+          "enabled.  Ignoring value %zu set in configuration file\n",
+          pilot_sc_group_size_, beam_block_size_);
+    }
+    beam_block_size_ = pilot_sc_group_size_;
```

Comment:
beam_block_size_ can also be allowed if it's a multiple of pilot_sc_group_size_.

Reply:
It is just a warning.  But we can remove the restriction on beam block size.

Reply:
Yes, we can remove the restriction since we have a general mechanism for start_sc. But set beam_block_size_ = pilot_sc_group_size_ as default in freq-orth-pilot for better efficiency (and default to 1 otherwise).

Reply:
Fixed